### PR TITLE
now version 1 is working as described in the readme

### PR DESCRIPTION
**Pull request by avogt-sundn**
your example had a bug in the ui (or i completely missed your point ;-) )
when the user updates, he needs to also send the received version. this was missing in the cliente.html.
I also refactored the ClienteResource class to use your dao

### CODE REVIEW

**Comment by avogt-sundn** [comment]
your example had a bug in the ui (or i completely missed your point ;-) )
when the user updates, he needs to also send the received version. this was missing in the cliente.html.
I also refactored the ClienteResource class to use your dao

**Comment by avogt-sundn** [comment]
and fixing the same issue on the cliente.html in the 2-database-concurrency-panache i only had to introduce the entityManager from Panache.getEntityManager() in order to call merge() to re-attach the detached cliente received via rest.

**Comment by manoelcampos** [comment]
Hello @avogt-sundn 

> your example had a bug in the ui (or i completely missed your point ;-) )
> when the user updates, he needs to also send the received version. this was missing in the cliente.html.

You're right about that. However, I've seen you included Lombok as a dependency. The intention of this repo is to reproduce an issue with the current version of Panache in Quarkus. So, the dependencies are expected to be kept minimal. Furthermore, Lombok requires IDE support to work.

> I also refactored the ClienteResource class to use your dao
Yeah, I forgot that. Thanks.
> and fixing the same issue on the cliente.html in the 2-database-concurrency-panache i only had to introduce the entityManager from Panache.getEntityManager() in order to call merge() to re-attach the detached cliente received via rest.

Yeah, I know. But that is a workaround, as discussed here [#7193](https://github.com/quarkusio/quarkus/issues/7193). And the second example is to show it's expected to work. The example is even linked in the issue above. So, the intention here is to keep as it is until a final solution is provided by Quarkus.

So, if you remove Lombok and keep just the changes in the first example, I merge your PR.
The Quarkus version shouldn't be changed either, in order to show in which version I found the issue.
Thanks for contributing.

**Comment by avogt-sundn** [comment]
Hey @manoelcampos you are right, so i removed lombok. You should fix the cliente.html versao bug - otherwise your failing example fails for the wrong reason. Regarding the panache situation which you want to point out: i added the correct entityManager.merge call to the example to make it run. This will help anybody the likes me coming around looking for the right answer to find it. It also documents the conversation you had in the bug https://github.com/quarkusio/quarkus/issues/7193 where everyone wonders why there is no merge() on PanacheEntity. Right?

And let me thank you for your effort - it helped me tremendously to proof that optimistic locking does work in Quarkus!

File: 1-database-concurrency-plain-hibernate/pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <maven.compiler.target>1.8</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <quarkus.version>1.2.0.Final</quarkus.version>
```
Comment (manoelcampos) [inline 1-database-concurrency-plain-hibernate/pom.xml:13]: Keep 1.2.0.Final because it's the version I found out the issue.

**Comment by manoelcampos** [comment]
Hello @avogt-sundn 
Thanks for your contribution to keep the discussion going.
I started a review with some comments.

File: 2-database-concurrency-panache/src/main/java/com/manoelcampos/server/rest/ClienteResource.java [hunk 2]
```diff
@@ -36,20 +45,20 @@ public long insert(Cliente cliente) {
...
     public void update(Cliente cliente) {
-        try{
-            if(Cliente.update(cliente))
-                return;
-        }catch(OptimisticLockException e){
+        try {
+            /**
+             * this is weird, why no CLiente.merge(cliente) ? Missing in the API
+             */
+            Panache.getEntityManager().merge(cliente);
```
Comment (manoelcampos) [inline 2-database-concurrency-panache/src/main/java/com/manoelcampos/server/rest/ClienteResource.java:52]: I think you missed the point.
Getting a direct instance to EntityManager, you're using just plain Hibernate and that sure works. 
`Panache.getEntityManager().merge(cliente)` is not the way it's supposed to work using only the  Panache interfaces. If you check the discussion on the quarkus repo's issue, it's explained why the `merge()` method is not directly exposed on Panache.

**Review by manoelcampos — CHANGES_REQUESTED** [review]

**Pull request closed without merging**